src/device-registry/utils/generate-filter.js

```diff
@@ -690,6 +698,12 @@ const generateFilter = {
       filter["_id"] = ObjectId(_id);
     }
 
+    if (site_codes) {
+      let siteCodesArray = site_codes.split(",");
```

Comment:
Allow user to provide a comma separated list of device codes in the query parameter for `site_codes` when querying for device details.

All matches will be returned.